Framework/DataHandling/src/SetSample.cpp

```diff
@@ -556,18 +586,34 @@ SetSample::createCylinderLikeXML(const Kernel::PropertyManager &args,
                  [](double val) { return val *= 0.01; });
   // XML needs center position of bottom base but user specifies center of
   // cylinder
-  const unsigned axisIdx = static_cast<unsigned>(refFrame.pointingUp());
-  const V3D baseCentre = cylBaseCentre(centre, height, axisIdx);
+  V3D baseCentre;
+  std::ostringstream XMLString;
+  if (args.existsProperty(ShapeArgs::AXIS)) {
+    try {
+      int axisInt = std::stoi(args.getPropertyValue(ShapeArgs::AXIS));
```

Comment:
Can I ask which part throws the `std::runtime_error` here?

Reply:
I just realised that the code doesn't work as expected. I have pushed a fix that does not use the try-catch

Reply:
Thanks for looking into this. Do we know why the tests didn't fail if it wasn't working as expected?

Reply:
That's because using `std::stoi` on `"1"` or `"0,1,0"` will always give an `int` and no runtime error would be thrown.

Reply:
Thanks for the explanation.